### DISCUSSION TITLE
update the customer's email

### DISCUSSION THREAD

**Discussion by Umut-Schweiz** [duncanmcclean/simple-commerce#934] [Help] [answered]
Hi guys, is it possible to update the customer email instead of creating a new one ?

https://simple-commerce.duncanmcclean.com/tags/cart -> Update Cart
I tried the relevant section in the document, but only those other than email work.

<input type="text" name="customer[name]">
<input type="email" name="customer[email]">

In this case, only name or however you define it is updated. But the email is not updated!

Actually, the only thing I want to ask is how can I update the customer's email in the cart?

**Comment by duncanmcclean** [comment, marked as answer]
Hey 👋

I've just did a little bit of testing and unfortunately, there's no way to update a customer's email address with the `{{ sc:cart:update }}` tag. When you provide a new email, it'll create a completely new customer. 

I can't change this without introducing a breaking change but I'll take note for v6. We can probably make it so `<input type="email" name="customer[email]">`  updates the existing customer entry if the order already has a customer set.

In the meantime, there's two ways you could potentially workaround this limitation:

* Use the [`{{ sc:customer:update }}`](https://simple-commerce.duncanmcclean.com/tags/customer#content-update-a-customer) tag (although, you can't use this inside another form)
* Use a placeholder field, listen for the order being updated, then update the customer
    * Use another name for the email field that's not picked up by Simple Commerce (eg `customer_email`) `<input type="email" name="customer_email">`
    * Add that field to the whitelist in your `simple-commerce.php`
    * Create an event listener that listens for the `OrderSaved` event. In that listener, you can check if `customer_email` is set on the order.
        * If the order doesn't already have a customer assigned, create one and associate it with the order. 
            * `Customer::make()->email($order->get('customer_email'))->save()`
            * `$order->customer($customer)->save()`
        * If the order does already have a customer, check if the email matches. If it doesn't, update the existing customer.
            * `$order->customer()->email($order->get('customer_email'))->save()`

Hopefully that's helpful - sorry it's not possible yet in SC!

**Reply by Umut-Schweiz** [reply]
Hi 👋

Thanks for your answer. A solution like this came to my mind. But it would be really easy to do this with {{ sc:customer:update }}.I will find a temporary solution for this and wait for the update :)
